src/app/page.tsx

```diff
@@ -1,47 +1,53 @@
-import Image from "next/image";
+"use client";
+
+import { PairsList } from "./PairsList";
+import { createContext, useEffect, useState } from "react";
+import * as adex from "alphadex-sdk-js";
+import { PairInfo } from "./PairInfo";
+
+export const AdexStateContext = createContext(null);
```

Comment:
Add this line just below the moved adex.init() line: 
let initial_staticState = new adex.StaticState(adex.clientState.internalState);

This sets up an initial StaticState variable that we can then use in the context and useState variables. If we dont do this, I get a type checking error because clientState is of type ClintState and the variable we get from the stateChanged$ subscription is StaticState. 

Then change the AdexStateContext line to:
export const AdexStateContext = createContext(initial_staticState);
This just sets the intial value of the context variable to a type StaticType which is the same type we will use to update it later.